### PR TITLE
PLT-5853 Contract store garbage collector

### PR DESCRIPTION
**Pull request by jhbertra**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by jhbertra** [comment]
@yveshauser let me know if you'd like to do a pair review on this.

File: marlowe-integration/src/Test/Integration/Marlowe/Local.hs [hunk 1]
```diff
@@ -264,6 +264,7 @@ withLocalMarloweRuntime' MarloweRuntimeOptions{..} test = withRunInIO \runInIO -
           { contractStoreDirectory = resolveWorkspacePath workspace "contract-store"
           , contractStoreStagingDirectory = resolveWorkspacePath workspace "contract-staging-area"
           , lockingMicrosecondsBetweenRetries = 100_000
+          , minContractAge = 60
```
Comment (bwbush) [inline marlowe-integration/src/Test/Integration/Marlowe/Local.hs:267]: Could we annotate this line with a comment about the time units?

**Review by bwbush — APPROVED** [review]
Looks good.

I'm wondering if it would have been good to have garbage collection triggered by the total size of contracts stored. IPFS, for example, has a storage limit and a high-water mark for garbage collection. It will throw a fatal error if the maximum storage is reached and there is nothing to collect garbage.  This provides natural backpressure. As it stands, is `marlowe-contract` susceptible to a DoS attack that would bring the whole Runtime down by filling the disk volume?

**Comment by jhbertra** [comment]
That's a great idea. I can change the implementation so that this happens.

Regarding susceptibility to DoS attacks, yes `marlowe-contract` is susceptible to this. It wouldn't bring down the whole runtime if the different components were isolated on separate machines, but it could bring down the ability to write more contracts.

**Pull request merged**